spice_agent/hardware/actions.py

```diff
@@ -44,6 +47,137 @@ def get_darwin_system_profiler_values(self) -> Dict[str, str]:
             "apple_serial_number": data_type.get("serial_number"),
         }
 
+    def _is_nvidia_smi_available(self) -> bool:
+        """
+        Checks if nvidia-smi is avaialble
+        """
+        try:
+            nvidia_smi_command = "nvidia-smi"
+            subprocess.check_output(nvidia_smi_command)
+            return True
+        except (FileNotFoundError, subprocess.CalledProcessError):
```

Comment:
Interestingly enough it does, but I don't have to use the sys call anymore since I'm using shutil which is nice:

```
>>> import subprocess
>>> try:
...     subprocess.check_output("nvidia-smii")
... except Exception as exception:
...     print(type(exception))
...
<class 'FileNotFoundError'>
```